lib/pages/media_view.dart

```diff
@@ -85,7 +93,35 @@ class MediaViewPage extends HookWidget {
                 IconButton(
                   icon: const Icon(Icons.file_download),
                   tooltip: 'download',
-                  onPressed: notImplemented,
+                  onPressed: () async {
+                    final File file =
+                        await DefaultCacheManager().getSingleFile(url);
+
+                    if ((Platform.isAndroid || Platform.isIOS) &&
+                        await requestMediaPermission()) {
```

Comment:
Do we give some sort of message if the user denied this permission?
Especially since Android tends to remember previous choices, it might look like nothing is happening.